R/config.R

```diff
@@ -185,48 +172,203 @@ print.snowflake_redacted <- function(x, ...) {
   cat(cli::col_grey("<REDACTED>"))
 }
 
+
+#' Load Snowflake configuration from all available sources
+#'
+#' Loads and consolidates Snowflake configuration from multiple sources with proper
+#' precedence handling. Configuration sources include:
+#'
+#' 1. `connections.toml` and `config.toml` files
+#' 2. Environment variables (both generic and connection-specific)
+#' 3. User-provided parameters
+#'
+#' The function follows Snowflake's client behavior for resolving connection names
+#' and handling conflicts between configuration sources.
+#'
+#' @param name A named connection to use. If NULL, the default connection will be used
+#' @param config_dir The directory containing Snowflake configuration files.
+#'   Defaults to the result of `default_config_dir()`.
+#'
+#' @return A list with the following components:
+#'   \item{connections}{List of available connection configurations}
+#'   \item{connection_name}{The resolved connection name to use}
+#'   \item{connection_file}{Path to the configuration file that was loaded}
+#'
+#' @keywords internal
+load_config <- function(name = NULL, config_dir = default_config_dir()) {
+  # Initialize result structure
+  result <- list(
+    connections = list(),
+    connection_name = NULL,
+    connection_file = NULL
+  )
+
+  # File paths
+  config_toml <- file.path(config_dir, "config.toml")
+  connections_toml <- file.path(config_dir, "connections.toml")
+
+  # Step 1: Resolve connection name with proper precedence:
+  # 1. Explicit name parameter
+  # 2. SNOWFLAKE_DEFAULT_CONNECTION_NAME environment variable
+  # 3. default_connection_name from config.toml
```

Comment:
In case this matters, I have `[connections.default]` in my `config.toml`. I do not have `default_connection_name` because the default connection name is "default".

https://github.com/snowflakedb/snowflake-connector-python/blob/539d67501cd29ed597e2e505a129e039ba08ad08/src/snowflake/connector/config_manager.py#L462-L465